### PR TITLE
Support new properties for azurerm_resource_group_template_deployment

### PR DESCRIPTION
**Pull request by neil-yechenwei**
fixes https://github.com/terraform-providers/terraform-provider-azurerm/issues/9718
fixes https://github.com/terraform-providers/terraform-provider-azurerm/issues/9719

--- PASS: TestAccResourceGroupTemplateDeployment_withOutputs (210.78s)
--- PASS: TestAccResourceGroupTemplateDeployment_incremental (219.87s)
--- PASS: TestAccResourceGroupTemplateDeployment_childItems (314.10s)
--- PASS: TestAccResourceGroupTemplateDeployment_singleItemUpdatingParams (338.39s)
--- PASS: TestAccResourceGroupTemplateDeployment_empty (338.57s)
--- PASS: TestAccResourceGroupTemplateDeployment_singleItemUpdatingTemplate (352.03s)
--- PASS: TestAccResourceGroupTemplateDeployment_multipleNestedItems (352.36s)
--- PASS: TestAccResourceGroupTemplateDeployment_multipleItems (360.59s)
--- PASS: TestAccResourceGroupTemplateDeployment_updateOnErrorDeployment (439.59s)
--- PASS: TestAccResourceGroupTemplateDeployment_switchBetweenParametersLinkAndParametersContent (599.27s)

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
@neil-yechenwei what's the use-case adding these fields is looking to solve? is there a Github issue for this?

**Comment by neil-yechenwei** [comment]
@tombuildsstuff , attached the github issues for new properties [`OnErrorDeployment`](https://github.com/terraform-providers/terraform-provider-azurerm/issues/9718), [`TemplateLink` and `ParametersLink`](https://github.com/terraform-providers/terraform-provider-azurerm/issues/9719).

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hi @neil-yechenwei 

Thanks for this PR.

I've taken a look through and left some comments inline, however after discussing this internally we've come to the conclusion that the `on_error_deployment` field doesn't make sense in Terraform.

In addition whilst we could look to support the `parameters_link` field here - we believe there's more downsides than upsides to doing so at this time, namely since this isn't likely to be reused across deployments, and may contain sensitive variables (which would have to be public, a huge security risk) - instead it'd make more sense to define these using the `parameters_content` against the deployment itself. That said, as mentioned in #9719, should a compelling use-case for this field become available we can take another look.

As such whilst I'd like to thank you for this contribution - since we believe these fields don't make sense in Terraform at this time I'm going to close this PR for the moment.

Thanks!

File: azurerm/internal/services/resource/resource_group_template_deployment_resource.go [hunk 1]
```diff
@@ -73,11 +73,59 @@ func resourceGroupTemplateDeploymentResource() *schema.Resource {
 				ValidateFunc: validation.StringInSlice(templateDeploymentDebugLevels, false),
 			},
 
+			"on_error_deployment": {
```
[the remark this answers is not in the document]
Comment (neil-yechenwei) [reply azurerm/internal/services/resource/resource_group_template_deployment_resource.go:76]: @tombuildsstuff , thanks for your comment. May I ask why it doesn't make sense in Terraform?

**Pull request closed without merging**